src/igvfd/audit/file_set.py

```diff
@@ -21,3 +21,160 @@ def audit_no_files(value, system):
+        audit_category: missing related seqspec files
+        audit_levels: ERROR
+    '''
+    if 'files' in value:
+        for file in value['files']:
+            if file.startswith('/sequence-files/'):
+                sequence_file_object = system.get('request').embed(file, '@@object?skip_calculated=true')
+
+                # Audit the file set with sequence files without the associated seqspec also in the file set.
+                if sequence_file_object.get('seqspec'):
+                    if sequence_file_object.get('seqspec') not in value['files']:
+                        seqspec_path = sequence_file_object['seqspec']
+                        detail = (
+                            f'File set {audit_link(path_to_text(value["@id"]), value["@id"])} has sequence file '
+                            f'{audit_link(path_to_text(file), file)} which links to seqspec '
+                            f'{audit_link(path_to_text(seqspec_path), seqspec_path)} which does not link to this file set.'
+                        )
+                        yield AuditFailure('missing related seqspec files', detail, level='ERROR')
+
+            # Audit the file set with a seqspec configuration file without the associated sequence files also in the file set.
+            if file.startswith('/configuration-files/'):
+                configuration_file_object = system.get('request').embed(file)
+                if configuration_file_object['content_type'] == 'seqspec' and configuration_file_object.get('seqspec_of'):
+                    missing_sequence_files = list(set(configuration_file_object.get(
+                        'seqspec_of', [])).difference(set(value['files'])))
+                    if missing_sequence_files:
+                        missing_sequence_files = ', '.join(
+                            [audit_link(path_to_text(sequence_file), sequence_file) for sequence_file in missing_sequence_files])
+                        detail = (
+                            f'File set {audit_link(path_to_text(value["@id"]), value["@id"])} has seqspec configuration file '
+                            f'{audit_link(path_to_text(file), file)} which links to sequence files: {missing_sequence_files} which '
+                            f'do not link to this file set.'
+                        )
+                        yield AuditFailure('missing related seqspec files', detail, level='ERROR')
+
+
+@audit_checker('FileSet', frame='object')
+def audit_inconsistent_seqspec(value, system):
+    '''
+        audit_detail: Sequence files in a file set from the same sequencing run and index are expected to link to the same seqspec file, which should be unique to that sequencing run and index.
+        audit_category: inconsistent seqspec metadata
+        audit_levels: ERROR
+    '''
+    if 'files' in value:
+        sequencing_runs = {}
+        indices = {}
+        for file in value['files']:
+            if file.startswith('/sequence-files/'):
```

Comment:
maybe for code readability, it would be better to define some kind of dictionary that collects all the possible sequencing files unique combinations of index, seq_run, etc.?)